src/main/java/duke/command/AddDeadlineCommand.java

```diff
@@ -0,0 +1,40 @@
+package duke.command;
+
+import java.time.format.DateTimeParseException;
+import duke.task.Deadline;
+import duke.task.Task;
+
+/**
+ * Adds a deadline task to the list of user's tasks.
+ */
+public class AddDeadlineCommand extends AddCommand {
+    public static final String COMMAND_IDENTIFIER = "deadline";
+
+    /**
+     * Returns the AddDeadline command represented by the user input.
+     *
+     * @param userInput String input provided by the user.
+     * @return AddDeadline user command.
+     * @throws MalformedCommandException If userInput is incorrectly formatted for an AddDeadline command.
+     */
+    public static Command create(String userInput) throws MalformedCommandException {
+        try {
+            String userParams = userInput.split(" ", 2)[1];
+            String[] userParamsSplit = userParams.split(" /", 2);
+            String description = userParamsSplit[0];
```

Comment:
You are using these code in AddEventCommand as well. Maybe you make it into one of Parser's functions?